tests/unit/tree/endpoint_node/test_propagate_spec_vars.py

```diff
@@ -0,0 +1,29 @@
+from pytest import mark
```

Comment:
@gabrieltron can we have a test in which we have vars with the same name for both parent and child? To check that the inner value should be the one returned: the child's one.

For example:


```yaml
endpoints:
  - name: snippets-api
    path: http://demo.scanapi.dev/api/v1/
    headers:
      Content-Type: application/json
    vars:
      my_var: one
    requests:
      - name: health
        path: health?${my_var}
    endpoints:
      - name: endpoint
        vars:
          my_var: two
        requests:
          - name: another health
            path: health?${my_var}
```

Request one should be: `GET http://demo.scanapi.dev/api/v1/health?one`
Request two should be: `GET http://demo.scanapi.dev/api/v1/health?two`

Reply:
Nice, I hadn't thought of that scenario. I changed the method so it only propagates the new vars.

I also changed it so it only writes if the variable doesn't exist. On the downside, this implies that there's no way of update global variables. On the upside, it further protects the variables already declared in the parents. Do you think this is a good approach? Or should request nodes be able to update global variables?